apisix/plugins/grpc-transcode/response.lua

```diff
@@ -14,18 +14,89 @@
 -- See the License for the specific language governing permissions and
 -- limitations under the License.
 --
-local util   = require("apisix.plugins.grpc-transcode.util")
+local util        = require("apisix.plugins.grpc-transcode.util")
+local grpc_proto  = require("apisix.plugins.grpc-transcode.proto")
 local core   = require("apisix.core")
 local pb     = require("pb")
 local ngx    = ngx
 local string = string
+local ngx_decode_base64 = ngx.decode_base64
+local ipairs = ipairs
+local pcall  = pcall
 
-return function(ctx, proto, service, method, pb_option)
+
+local function handle_error_response(status_detail_type)
+    local grpc_status = ngx.header["grpc-status-details-bin"]
+    if grpc_status then
+        grpc_status = ngx_decode_base64(grpc_status)
+        if grpc_status == nil then
+            ngx.arg[1] = "grpc-status-details-bin is not base64 format"
+            return "grpc-status-details-bin is not base64 format"
```

Comment:
We can exact the error message to a variable so we don't need to repeat them?

Reply:
Done